Reduce unsafeness in FetchEvent class

annevk opened the pull request:
#### 4a66bebe11889cad351d1aab4295d1b57ac37c7a
<pre>
Reduce unsafeness in FetchEvent class
<a href="https://bugs.webkit.org/show_bug.cgi?id=293615">https://bugs.webkit.org/show_bug.cgi?id=293615</a>
<a href="https://rdar.apple.com/152083443">rdar://152083443</a>

Reviewed by NOBODY (OOPS!).

Apply https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4a66bebe11889cad351d1aab4295d1b57ac37c7a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/105332 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/25044 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/15471 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/110540 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/55993 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/107373 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/25484 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/33587 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80004 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/55993 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/108338 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/19879 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95069 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60310 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/19636 "") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/13145 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/55384 "Built successfully") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/89355 "") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/13188 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/113199 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/32489 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/23946 "") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89080 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/32852 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/91286 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/88720 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/33614 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/11401 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/27966 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17090 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/32412 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/37825 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/32185 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/35531 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/33759 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/workers/service/FetchEvent.cpp:
```diff
@@ -90,10 +90,11 @@ ExceptionOr<void> FetchEvent::respondWith(Ref<DOMPromise>&& promise)
 
-    m_respondPromise = WTFMove(promise);
-    addExtendLifetimePromise(*m_respondPromise);
+    Ref respondPromise = WTFMove(promise);
```
We don't really need `respondPromise`, we can just use `promise`

annevk replied:
Ah, nice!

cdumez commented on Source/WebCore/workers/service/FetchEvent.cpp:
```diff
@@ -90,10 +90,11 @@ ExceptionOr<void> FetchEvent::respondWith(Ref<DOMPromise>&& promise)
-    m_respondPromise = WTFMove(promise);
-    addExtendLifetimePromise(*m_respondPromise);
+    Ref respondPromise = WTFMove(promise);
+    m_respondPromise = respondPromise.copyRef();
```
`m_respondPromise = promise.copyRef();`

cdumez commented on Source/WebCore/workers/service/FetchEvent.cpp:
```diff
@@ -90,10 +90,11 @@ ExceptionOr<void> FetchEvent::respondWith(Ref<DOMPromise>&& promise)
-    addExtendLifetimePromise(*m_respondPromise);
+    Ref respondPromise = WTFMove(promise);
+    m_respondPromise = respondPromise.copyRef();
+    addExtendLifetimePromise(respondPromise.get());
```
`addExtendLifetimePromise(promise.get());`

cdumez commented on Source/WebCore/workers/service/FetchEvent.cpp:
```diff
@@ -90,10 +90,11 @@ ExceptionOr<void> FetchEvent::respondWith(Ref<DOMPromise>&& promise)
+    addExtendLifetimePromise(respondPromise.get());
 
-    auto isRegistered = m_respondPromise->whenSettled([protectedThis = Ref { *this }] {
+    auto isRegistered = respondPromise->whenSettled([protectedThis = Ref { *this }] {
```
`promise->`

cdumez reviewed (approved):
LGTM with nit.

annevk committed:
Reduce unsafeness in FetchEvent class
https://bugs.webkit.org/show_bug.cgi?id=293615
rdar://152083443

Reviewed by Chris Dumez.

Apply https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines

Canonical link: https://commits.webkit.org/295533@main
